docs/world_generation.md

```diff
@@ -0,0 +1,75 @@
+# World generation
+World generation in Hypermine is constrained the following principles:
+* The generated contents of each chunk must depend only on the parameters of the world generation algorithm, not on gameplay.
+* Everything must be generated in its settled state to ensure that unmodified terrain does not need to be stored on disk, and to help with immersion.
+* World generation should be as isotropic as reasonably possible, so that it is not obvious which direction leads to the origin.
+
+These constraints help inspire the details of the world generation algorithm, which are explained further in the sections below.
+
+## Noise
+Every procedural world generation algorithm needs to start with some way of generating noise. This noise is useful for determining properties like temperature and rainfall (which affects what material the ground is made of), and it is also used in determining the shape of the terrain. While Perlin noise is often used as a standard approach, it is not clear how it would be adapted to a large hyperbolic world. Instead, Hypermine uses a different approach specifically designed for a hyperbolic tiling, inspired by Hyperrogue.
+
+### Coarse noise function output
+The first step is to form a coarse approximation of the noise function, deciding on one value for each node. To do this, we take advantage of the tiling itself. For a 2D analogy, the pentagonal tiling of the hyperbolic plane can be thought of as a set of lines dividing the hyperbolic plane instead of individual pentagons.
+
+TODO: Picture of pentagonal tiling with lines colored to distinguish them from each other
+
+Similarly, in 3D, the dodecahedral tiling can be thought of as a set of planes dividing hyperbolic space. This interpretation of the dodecahedral tiling is important for understanding how the noise function works between nodes.
+
+To decide on a noise value for each node, we break the dodecahedral tiling up into this planes. We associate each plane with a randomly chosen noise value offset, such that crossing a specific plane in one direction increases or decreases the noise value by a specific amount, and crossing the same plane from the other side has the opposite effect. Once we decide on a noise value for the root node, this definition fully determines the noise value of every other node.
```

Comment:
editorial: 
- "into this planes" is probably not what you intended.
- "value offset" feels like reinventing the term "delta" more obscurely, but maybe that's just me

Reply:
First bullet point: Whoops. Good catch. This is why peer review is important (I proofread it, but I think I just read it as "into these planes", which is what I should have written)

Second bullet point: I like the idea of "delta". In an earlier draft, I had thought of using "temperature" as a concrete example of the noise function, but after I wrote out more of the docs, I found that it wasn't as helpful as I expected, so I switched from the word "temperature" to "noise value". However, I do like the brevity of "noise delta" instead of "noise value offset".